Xamarin.PropertyEditing/ViewModels/PropertyViewModel.cs

```diff
@@ -480,24 +479,32 @@ private async void UpdateAutocomplete (string value)
 					tasks.Add (complete.GetCompletionsAsync (Property, value, cancel));
 				}
 
-				IReadOnlyList<string> list = null;
-				do {
-					Task<IReadOnlyList<string>> results = await Task.WhenAny (tasks);
-					tasks.Remove (results);
-
-					if (list == null) {
-						list = await results;
-						common = new HashSet<string> (list);
-					} else
-						common.IntersectWith (await results);
-				} while (tasks.Count > 0 && !cancel.IsCancellationRequested);
-
-				cancel.ThrowIfCancellationRequested ();
-				this.autocomplete.Reset (list.Where (common.Contains));
+				await ResetAutocompleteCollection (this.autocomplete, tasks, cancel);
 			} catch (OperationCanceledException) {
 			}
 		}
 
+		public async Task ResetAutocompleteCollection (ObservableCollectionEx<string> collection, List<Task<IReadOnlyList<string>>> tasks, CancellationToken cancel)
```

Comment:
@ermau Not sure if this is the sort of thing you had in mind, but extracted this out to make it easier to isolate this functionality.

Reply:
Not sure why this needed to be extracted (you don't need it to test the behavior), let alone public. What external consumer would ever invoke this, especially with these arguments?

Reply:
Making sure the exception is thrown is _not_ the thing to be testing here, it's the completion results when the results finished, but it was canceled.

Reply:
As your comment was literally on the newly added `cancel.ThrowIfCancellationRequested ();` I thought we wanted to ensure the exception was thrown, if the cancelation token's `IsCancelationRequested` was true. 

OK, So just to double check... We want to test that AutocompletionItems has *not* been updated when `GetCompletionsAsync` is cancelled??

If I'm still going down the wrong path, maybe a Slack call is in order?

Reply:
> As your comment was literally on the newly added cancel.ThrowIfCancellationRequested (); I thought we wanted to ensure the exception was thrown, if the cancelation token's IsCancelationRequested was true.

There's an expected output, the contents of the autocomplete list. Exiting the method before the contents can be changed via the reset (by using `cancel.ThrowIfCancellationRequested()`) changes that expected output. 

> OK, So just to double check... We want to test that AutocompletionItems has not been updated when GetCompletionsAsync is cancelled??

Yes by way of the cancellation that's triggered if they start entering more text before the previous autocomplete requests can finish. Here's a demonstration in the change in behavior, in a case where you have 2 object editors.

1. First object editor's `GetCompletionAsync()` completes and returns items.
2. New text is entered, cancelling the second request and exiting the loop.
3. The list resets with the contents of the aggregate list, which will only contain items from 1.

With the throw before reset, #3 no longer occurs, living the auto-complete list to be whatever it was previously. That's the change in behavior to be tested for.